### ISSUE TITLE
[Feature Request] S3 Compatibility

### ISSUE THREAD

**Issue by sanskar-mk2** [axeloz/filesharing#59]
Many cloud filesystem are s3 compatible, if we can do it, then we can have more ways to store file. I am working on it. Do you have suggestions?

**Comment by axeloz** [comment]
Hi @sanskar-mk2, you want to add the S3 support for Filesharing? Is that right?

**Comment by sanskar-mk2** [comment]
Yes, I have a working S3 compatible (I am using Digital Ocean Spaces) version of the app. Are you working on any such plans right now? If not, I will be happy to PR it.

Additionally I have also made some quality of life updates to it.

**Comment by axeloz** [comment]
No, I had no plan on implementing S3 but with Flysystem on Laravel, it's pretty straightforward. Question is: is there a need for this?

**Comment by sanskar-mk2** [comment]
Yes, there is need for it, since the filesystem is hardcoded in the controllers. And to create a zip file, you cannot write directly to S3. Some little changes are needed.

**Comment by axeloz** [comment]
I was talking about the functional need, a user request. But if you think some users may need to change the disk driver, then you are right: some improvements are required.

**Comment by sanskar-mk2** [comment]
Yes, people use different disk driver. I need it so I can buy compute and storage separately.
